src/roles/roles.py

```diff
@@ -16,3 +22,78 @@ def get_roles_to_assign(address: str) -> List[str]:
 
 async def assign_roles(member: discord.Member, roles: List[discord.Role]):
     await member.add_roles(*roles)
+
+
+async def unassign_roles(member: discord.Member, roles: List[discord.Role]):
+    await member.remove_roles(*roles)
+
+
+@dataclass(frozen=True)
+class DiscordUser:
+    guild: discord.Guild
+    member: discord.Member
+    proposed_roles: List[discord.Role]
+
+
+def get_role(guild: discord.Guild, role_name: str) -> discord.Role:
+    role: Optional[discord.Role] = get(guild.roles, name=role_name)
+    if role is None:
+        raise Exception(f"Role {role_name} does not exist.")
+    return role
+
+
+def get_discord_user(guild_id: str, user_id: str, address: str) -> DiscordUser:
+    _guild: Optional[discord.Guild] = get_guild(guild_id)
+    if _guild is None:
+        raise Exception("Guild does not exist.")
+    guild: discord.Guild = _guild
+
+    _member: Optional[discord.Member] = get_member(user_id, guild)
+    if _member is None:
+        raise Exception("Member does not exist in guild.")
+    member: discord.Member = _member
+
+    roles = get_roles_to_assign(address)
+    discord_roles: List[discord.Role] = []
+
+    if len(roles) > 0:
+        discord_roles = list(
+            map(lambda role_name: get_role(guild, role_name), roles))
+
+    return DiscordUser(guild, member, discord_roles)
+
+
+@loop(hours=1)
```

Comment:
seems like the `reconnect` kwarg for this decorator is set to True by default. Which means the task will restart on exception raise, right?

Reply:
Since I catch the exception in the for loop, it shouldn't reach the top of the function.

Reply:
ah didn't notice the try/except. Sweet